rpc/src/module/chain.rs

```diff
@@ -118,6 +121,17 @@ impl<CS: ChainStore + 'static> ChainRpc for ChainRpcImpl<CS> {
             .expect("current_epoch exists"))
     }
 
+    fn get_epoch_by_number(&self, number: String) -> Result<Option<EpochExt>> {
+        Ok(self
+            .shared
+            .block_hash(
+                number
+                    .parse::<BlockNumber>()
+                    .map_err(|_| Error::parse_error())?,
+            )
+            .and_then(|hash| self.shared.get_epoch_ext(&hash).map(Into::into)))
+    }
+
```

Comment:
It's incorrect implementation.
Currently, there is no ready-made epoch_number-> epoch_ext index.
First step, add the index, then leverage it in RPC.

Reply:
I've made fixes accordingly:

* Create database entry `epoch_number => epoch_index`
* refactor `ChainStore::get_epoch_ext` to receive epoch index as parameter instead
* add `ChainStore::get_epoch_index`
* add `ChainStore::get_block_epoch_index`

PTAL